### PR TITLE
Ports Space Dragon rework

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

Makes Space Dragon antag be actually up to date, since yogstation doesn't have [the rework](https://github.com/tgstation/tgstation/commit/422ef20ba8a6a754aa06d9ad0ca1913e867b14fd). This rework gives the dragon a mission to open and protect 3 carp rifts, and not just be a sentient megafauna that arrived on the station to kill the heads. 

Also my first PR with this had merge issues, so i create this one.

also it ports this https://github.com/tgstation/tgstation/pull/61539

# Wiki Documentation

Actually yogstation wiki doesn't have a page about this antagonist, so it should be created. 

# Changelog

:cl:  Authors: IndieanaJones, ExcessiveUseOfCobblestone, ported by SuperSlayer
rscadd: Ports Space Dragon rework from tg
imageadd: Adds new fancy images for the Space Dragon and for the carp rift.  
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1 | HEAR_1
+	melee_damage_upper = 35
+	melee_damage_lower = 35
+	armour_penetration = 30
+	pixel_x = -16
+	turns_per_move = 5
+	spacewalk = TRUE
+	spacewalk = TRUE
+	spacewalk = TRUE
```
Comment (tattax) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]:
```suggestion
	spacewalk = TRUE
```

Comment (ghost) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm]: Fixed

**Comment by ghost** [comment]
Checks please

**Review by nesisknesis — APPROVED** [review]

**Comment by ghost** [comment]
Cheeeeeecks

**Review by nmajask — CHANGES_REQUESTED** [review]
Credit the original author of the PR

**Comment by ghost** [comment]
> Credit the original author of the PR

Credited

**Comment by ghost** [comment]
@nmajask

File: code/modules/events/space_dragon.dm [hunk 2]
```diff
@@ -3,20 +3,19 @@
 	typepath = /datum/round_event/ghost_role/space_dragon
 	max_occurrences = 1
 	weight = 8
-	earliest_start = 70 MINUTES
-	min_players = 30
-	dynamic_should_hijack = TRUE
+	weight = 5
+	earliest_start = 35 MINUTES
```
Comment (JamieD1) [inline code/modules/events/space_dragon.dm]:
```suggestion
	weight = 5
	earliest_start = 35 MINUTES

```

Please dont remove our Min_Players and earliest start.

Comment (ghost) [reply code/modules/events/space_dragon.dm]: Uhhh, what do you mean?

Comment (ghost) [reply code/modules/events/space_dragon.dm]: Ah, okay

File: code/modules/events/space_dragon.dm [hunk 3]
```diff
@@ -2,21 +2,20 @@
 	name = "Spawn Space Dragon"
 	typepath = /datum/round_event/ghost_role/space_dragon
 	max_occurrences = 1
-	weight = 8
-	earliest_start = 70 MINUTES
+	weight = 5
+	earliest_start = 35 MINUTES
```
Comment (JamieD1) [inline code/modules/events/space_dragon.dm]:
```suggestion
	earliest_start = 60 MINUTES
```

**Comment by JamieD1** [comment]
Test merging.

**Comment by SegaCD** [comment]
Butchering the space dragon does not yield carp dragon hide.

**Comment by ghost** [comment]
Hmmmm

**Comment by tattax** [comment]
add the correct mutation colors for the dragon as to not make it grey everytime

**Comment by ghost** [comment]
> add the correct mutation colors for the dragon as to not make it grey everytime

Okay, i would try to do this, but this unfortunately wouldn't be soon(maybe)

**Comment by JamieD1** [comment]
> add the correct mutation colors for the dragon as to not make it grey everytime

This please, I dont want it to be boring grey.

**Comment by ghost** [comment]
> This please, I dont want it to be boring grey.

I don't know how to do this

**Comment by Marmio64** [comment]
this probably removed the carp dragon scales i added a while back

**Comment by ghost** [comment]
> this probably removed the carp dragon scales i added a while back

What path do they have? I can return them as a butcher result if needed.

**Comment by ghost** [comment]
Fuck

**Comment by ghost** [comment]
I hate github

**Comment by ghost** [comment]
Okay, because github shows me not existing errors I can’t implement the color update

**Comment by ghost** [comment]
???
CD890C41-9009-4842-97D5-0EE9C93159C0

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+	return TRUE
+
+/**
+   * Allows space dragon to choose its own name.
+   *
+   * Prompts the space dragon to choose a name, which it will then apply to itself.
+   * If the name is invalid, will re-prompt the dragon until a proper name is chosen.
+   */
+ /mob/living/simple_animal/hostile/space_dragon/proc/dragon_name()
+ 	var/chosen_name = sanitize_name(reject_bad_text(stripped_input(src, "What would you like your name to be?", "Choose Your Name", real_name, MAX_NAME_LEN)))
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]:
```suggestion
 	var/chosen_name = reject_bad_text(stripped_input(src, "What would you like your name to be?", "Choose Your Name", real_name, MAX_NAME_LEN))
```

**Comment by Marmio64** [comment]
> > this probably removed the carp dragon scales i added a while back
> 
> What path do they have? I can return them as a butcher result if needed.

just look at the code of the current mob, since this code hasnt modified the server yet

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+	flags_1 = PREVENT_CONTENTS_EXPLOSION_1 | HEAR_1
+	melee_damage_upper = 35
+	melee_damage_lower = 35
+	armour_penetration = 30
+	pixel_x = -16
+	turns_per_move = 5
+	spacewalk = TRUE
+	ranged = TRUE
+	mouse_opacity = MOUSE_OPACITY_ICON
+	butcher_results = list(/obj/item/stack/ore/diamond = 5, /obj/item/stack/sheet/sinew = 5, /obj/item/stack/sheet/bone = 30)
```
Comment (JamieD1) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:47]:
```suggestion1
	melee_damage_lower = 35
	armour_penetration = 30
	pixel_x = -16
	turns_per_move = 5
	spacewalk = TRUE
	ranged = TRUE
	mouse_opacity = MOUSE_OPACITY_ICON
	butcher_results = list(/obj/item/stack/ore/diamond = 5, /obj/item/stack/sheet/sinew = 5, /obj/item/stack/sheet/bone = 30)
```

Comment (JamieD1) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:47]:
```suggestion
	melee_damage_upper = 35
	melee_damage_lower = 35
	armour_penetration = 30
	pixel_x = -16
	turns_per_move = 5
	spacewalk = TRUE
	ranged = TRUE
	mouse_opacity = MOUSE_OPACITY_ICON
	butcher_results = list(/obj/item/stack/ore/diamond = 5, /obj/item/stack/sheet/sinew = 5, /obj/item/stack/sheet/bone = 30)
```

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
+		var/datum/action/small_sprite/small_action = new small_sprite_type()
+		small_action.Grant(src)
+	gust = new
+	gust.Grant(src)
+	rift = new
+	rift.Grant(src)
+
+/mob/living/simple_animal/hostile/space_dragon/Login()
+ 	. = ..()
+ 	if(!chosen_color)
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]:
```suggestion
 	if(!picked_color)
```

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 7]
```diff
@@ -0,0 +1,614 @@
...
+ 	///if(!picked_color)
+ 	dragon_name()
+ 		///color_selection()
+
+/mob/living/simple_animal/hostile/space_dragon/Life(mapload)
+	. = ..()
+	tiredness = max(tiredness - 1, 0)
+	for(var/mob/living/consumed_mob in src)
+ 		if(consumed_mob.stat == DEAD)
+ 			 continue
```
Comment (ghost) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm]:
```suggestion
 			continue
```

**Comment by ghost** [comment]
97E1240F-8F75-4349-B58C-CE899F416B6E
IT FUCKING SAYS THAT I HAVE INCONSISTENT INDENTATIONS IN FUCKING COMMENTS WHAT IS WRONG WITH IT

**Comment by ghost** [comment]
Okay fuck this, i give up. until someone will not fix this github related shit I can’t literally do anything.

File: code/modules/antagonists/space_dragon/space_dragon.dm [hunk 8]
```diff
@@ -1,32 +1,51 @@
...
-	to_chat(owner, "<b>Fabulous secret powers were revealed to me the day I held aloft a wizard's staff of change and said 'By the power of Draco, I have the power!'</b>")
-	to_chat(owner, "<b>The wizard was turned into the short-lived Pastry Cat while I became Space Dragon, the most powerful beast in the universe.</b>")
-	to_chat(owner, "<b>Clicking a tile will shoot fire onto that tile.</b>")
-	to_chat(owner, "<b>Using Tail Sweep will let me get the better of those who come too close.</b>")
-	to_chat(owner, "<b>Attacking dead bodies will allow me to gib them to restore health.</b>")
-	to_chat(owner, "<b>From the wizard's writings, he had been studying this station and its hierarchy.  From this, I know who leads the station, and will kill them so the station underlings see me as their new leader.</b>")
+	to_chat(owner, "<b>Endless time and space we have moved through.  We do not remember from where we came, we do not know where we will go.  All space belongs to us.\n\
+					Space is an empty void, of which our kind is the apex predator, and there was little to rival our claim to this title.\n\
```
Comment (Anvilman6) [inline code/modules/antagonists/space_dragon/space_dragon.dm]:
```suggestion
					Space is an empty void, in which our kind is the apex predator, and there is little to rival our claim to this title.\n\
```

File: code/modules/antagonists/space_dragon/space_dragon.dm [hunk 9]
```diff
@@ -1,32 +1,51 @@
...
+					But now, we find intruders spread out amongst our claim, willing to fight our teeth with magics unimaginable, their dens like lights flicking in the depths of space.\n\
```
Comment (Anvilman6) [inline code/modules/antagonists/space_dragon/space_dragon.dm]:
```suggestion
					But now, we find intruders spread out amongst our claim! They are willing to fight our teeth with magics unimaginable, in their strange metal dens flickering in the depths of space.\n\
```

File: code/modules/antagonists/space_dragon/space_dragon.dm [hunk 10]
```diff
@@ -1,32 +1,51 @@
...
+					Today, we will snuff out one of those lights.</b>")
```
Comment (Anvilman6) [inline code/modules/antagonists/space_dragon/space_dragon.dm]:
```suggestion
					Today, we will remind them of our power.</b>")
```

File: code/modules/events/space_dragon.dm [hunk 11]
```diff
@@ -2,21 +2,20 @@
...
-	dynamic_should_hijack = TRUE
 
 /datum/round_event/ghost_role/space_dragon
 	minimum_required = 1
 	role_name = "Space Dragon"
 	announceWhen = 10
 
 /datum/round_event/ghost_role/space_dragon/announce(fake)
-	priority_announce("It appears a lifeform with magical traces is approaching [station_name()], please stand-by.", "Lifesign Alert")
+	priority_announce("A large organic energy flux has been recorded near of [station_name()], please stand-by.", "Lifesign Alert")
```
Comment (Anvilman6) [inline code/modules/events/space_dragon.dm]:
```suggestion
	priority_announce("A large organic energy flux has been recorded in the vicinity of [station_name()], please stand-by.", "Lifesign Alert")
```

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 12]
```diff
@@ -462,6 +462,52 @@
...
+	var/mob/living/simple_animal/hostile/space_dragon/S = new (pick(spawn_locs))
+	player_mind.transfer_to(S)
+	player_mind.assigned_role = "Space Dragon"
+	player_mind.special_role = "Space Dragon"
+	player_mind.add_antag_datum(/datum/antagonist/space_dragon)
+
+	playsound(S, 'sound/magic/ethereal_exit.ogg', 50, TRUE, -1)
+	message_admins("[ADMIN_LOOKUPFLW(S)] has been made into a Space Dragon by the midround ruleset.")
+	log_game("DYNAMIC: [key_name(S)] was spawned as a Space Dragon by the midround ruleset.")
+	priority_announce("A large organic energy flux has been recorded near of [station_name()], please stand-by.", "Lifesign Alert")
```
Comment (Anvilman6) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]:
```suggestion
	priority_announce("A large organic energy flux has been recorded in the vicinity of [station_name()], please stand-by.", "Lifesign Alert")
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
Grammar syntax review

**Review by JamieD1 — APPROVED** [review]
hopefully this works

**Pull request merged**